pennylane/devices/preprocess.py

```diff
@@ -200,7 +200,9 @@ def mid_circuit_measurements(
         return qml.dynamic_one_shot(tape, postselect_mode=mcm_config.postselect_mode)
     if mcm_method == "tree-traversal":
         return (tape,), null_postprocessing
-    return qml.defer_measurements(tape, device=device)
+    return qml.defer_measurements(
```

Comment:
I might actually propose we do a breaking change to `mid_circuit_measurements` to also change from accepting a device to accepting `allow_postselect`.   But we can do that in a follow up.

Reply:
This would be an acceptable breaking change.  It doesn't really render it the docs.